EssentialApp/EssentialApp/CommentsUIComposer.swift

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import UIKit
+import Combine
+import EssentialFeed
+import EssentialFeediOS
+public final class CommentsUIComposer {
+	private init() {}
+
+	private typealias CommentsPresentationAdapter = LoadResourcePresentationAdapter<[ImageComment], CommentsViewAdapter>
+
+	public static func commentsComposedWith(
+		loader: @escaping () -> AnyPublisher<[ImageComment], Error>
+	) -> ListViewController {
+		let presentationAdapter = CommentsPresentationAdapter(loader: loader)
+
+		let commentsController = makeCommentsViewController(title: ImageCommentsPresenter.title)
+		commentsController.onRefresh = presentationAdapter.loadResource
+
+		presentationAdapter.presenter = LoadResourcePresenter(
+			resourceView: CommentsViewAdapter(
+				controller: commentsController),
+			loadingView: WeakRefVirtualProxy(commentsController),
+			errorView: WeakRefVirtualProxy(commentsController),
+			mapper: { ImageCommentsPresenter.map($0) })
+
+		return commentsController
+	}
+
+	private static func makeCommentsViewController(title: String) -> ListViewController {
+		let bundle = Bundle(for: ListViewController.self)
+		let storyboard = UIStoryboard(name: "ImageComments", bundle: bundle)
+		let controller = storyboard.instantiateInitialViewController() as! ListViewController
+		controller.title = title
+		return controller
+	}
+}
+
+final class CommentsViewAdapter: ResourceView {
+	private weak var controller: ListViewController?
+
+	init(controller: ListViewController) {
+		self.controller = controller
+	}
+
+	public func display(_ viewModel: ImageCommentsViewModel) {
```

Comment:
This function doesn't need to be marked as `public`, since it's not supposed to be accessed outside of the main module.
```suggestion
	func display(_ viewModel: ImageCommentsViewModel) {
```